### DISCUSSION TITLE
Which is  "legacy behavior": `shouldRerenderOnTransaction: false` or "re-render[ing] the editor on each transaction"

### DISCUSSION THREAD

**Discussion by jordanwilking** [ueberdosis/tiptap#5553] [General Questions] [answered]
The comment for the `shouldRerenderOnTransaction` property says:

"This is legacy behavior that will be removed in future versions."

Does this refer to the property `shouldRerenderOnTransaction` or to the current behavior of rerendering on every transaction?

https://github.com/ueberdosis/tiptap/blob/9907eb32c248eaffb86e12ac3d4da44495335c97/packages/react/src/useEditor.ts#L34

The `shouldRerenderOnTransaction` property is listed in the [July 14 blog post](https://tiptap.dev/blog/release-notes/say-hello-to-tiptap-2-5-our-most-performant-editor-yet), so I'm assuming that the current, rerender on every transaction, behavior is the "legacy behavior".

---

For context, I'm asking because I want the performance benefits from setting `shouldRerenderOnTransaction` to `false`, but only if that behavior is here to stay.

**Comment by nperez0111** [comment, marked as answer]
Re rendering on every transaction in the default and legacy behavior. With V3 we will make it so that you have to read from the editor state with useEditorState hook to get re rendering based on state

**Reply by jordanwilking** [reply]
Sounds good. Thank you for clearing that up.
